src/id_proofs.rs

```diff
@@ -0,0 +1,241 @@
+    AttributeNotInSet {
+        #[serde(flatten)]
+        statement: AttributeNotInSetStatement<Tag, Value>,
+    },
+}
+
+/// Serves as a uniFFI compatible bridge to [`concordium_base::id::id_proof_types::AtomicStatement<ArCurve, AttributeTag, AttributeKind>`]
+pub type AtomicStatementV1 = AtomicStatement<AttributeTag, String>;
+
+#[derive(Serialize)]
+#[serde(transparent)]
+pub struct Statement<Tag: Clone, Value: Clone> {
+    pub statements: Vec<AtomicStatement<Tag, Value>>,
+}
+
+/// Serves as a uniFFI compatible bridge to [`concordium_base::id::id_proof_types::Statement<ArCurve, AttributeKind>`]
+pub type StatementV1 = Statement<AttributeTag, String>;
+
+impl TryFrom<StatementV1>
+    for concordium_base::id::id_proof_types::Statement<ArCurve, AttributeKind>
+{
+    type Error = serde_json::Error;
+
+    fn try_from(value: StatementV1) -> Result<Self, Self::Error> {
+        serde_json::to_string(&value).and_then(|s| serde_json::from_str(&s))
+    }
+}
+
+#[derive(Serialize, Deserialize)]
+#[serde(tag = "type")]
+pub enum AtomicProof<Value> {
+    /// The atomic statement stating that an attribute should be revealed.
+    RevealAttribute { attribute: Value, proof: Bytes },
+    /// The atomic proof stating that an attribute is in a range.
+    AttributeInRange { proof: Bytes },
+    /// The atomic proof stating that an attribute is in a set.
+    AttributeInSet { proof: Bytes },
+    /// The atomic proof stating that an attribute is not in a set.
+    AttributeNotInSet { proof: Bytes },
+}
+
+/// Serves as a uniFFI compatible bridge to [`concordium_base::id::id_proof_types::AtomicProof<ArCurve, AttributeKind>`]
+pub type AtomicProofV1 = AtomicProof<String>;
+
+#[derive(Deserialize)]
+pub struct Proof<Value> {
+    pub proofs: Vec<AtomicProof<Value>>,
+}
+
+/// Serves as a uniFFI compatible bridge to [`concordium_base::id::id_proof_types::Proof<ArCurve, AttributeKind>`]
+pub type ProofV1 = Proof<String>;
+
+/// Serves as a uniFFI compatible bridge to [`concordium_base::common::Versioned<concordium_base::id::id_proof_types::Proof<ArCurve, AttributeKind>>`]
+pub type VersionedProofV1 = Versioned<ProofV1>;
+
+impl
+    TryFrom<
+        concordium_base::common::Versioned<
+            concordium_base::id::id_proof_types::Proof<ArCurve, AttributeKind>,
+        >,
+    > for VersionedProofV1
+{
+    type Error = serde_json::Error;
+
+    fn try_from(
+        value: concordium_base::common::Versioned<
+            concordium_base::id::id_proof_types::Proof<ArCurve, AttributeKind>,
+        >,
+    ) -> Result<Self, Self::Error> {
+        serde_json::to_string(&value).and_then(|s| serde_json::from_str(&s))
+    }
+}
+
+/// Prove a [`StatementV1`] given the provided context, producing a [`VersionedProofV1`]
+#[allow(clippy::too_many_arguments)]
+pub fn prove_statement_v1(
+    seed: Bytes,
+    net: Network,
+    global_context: GlobalContext,
+    ip_index: u32,
+    identity_index: u32,
```

Comment:
Inspiration is mainly from JS SDK with regards to the input for this function, and in the corresponding input there it's called `identityIndex` 🤷 

I personally like `identity_index` more as well since it describes the "ip>identity>credential" path on the derivation function on the seed better.